spec/lib/msf/core/exploit/remote/remote/http/http_cookie_spec.rb

```diff
@@ -0,0 +1,696 @@
+        n = nil
+
+        cookie.expires = n
+        cookie.max_age = n
+
+        expect(cookie.expired?(Time.now)).to eql(false)
+      end
+
+      it 'before "cookie.expires" is passed to expired? will return false' do
+        t = Faker::Time.backward(days: 1)
+
+        cookie.expires = Time.now
+
+        expect(cookie.expired?(t)).to eq(false)
+      end
+
+      it 'after "cookie.expires" is passed to expired? will return true' do
+        t = Faker::Time.forward(days: 1)
+
+        cookie.expires = Time.now
+
+        expect(cookie.expired?(t)).to eql(true)
+      end
+
+      it 'passed to expired?, will return true when cookie.created_at + max_age.seconds is before the passed value' do
+        t = Faker::Time.forward(days: 1)
+
+        cookie.max_age = 1
+        cookie.created_at = Time.now
+
+        expect(cookie.expired?(t)).to eq(true)
+      end
+
+      it 'passed to expired?, will return false when created_at + seconds.max_age is after the passed value' do
+        t = Faker::Time.backward(days: 1)
+
+        cookie.max_age = 1
+        cookie.created_at = Time.now
+
+        expect(cookie.expired?(t)).to eq(false)
+      end
+    end
+  end
+
+  describe 'valid_for_uri?' do
+    it "will return false when domain hasn't been set" do
+      # domain set as nil in before
+
+      expect do
+        cookie.valid_for_uri?(URI(random_string))
+      end.to raise_error(ArgumentError)
+    end
+
+    it 'will return false if secure is set as true and the passed cookie is a https url' do
+      cookie.domain = random_string
+
+      v = cookie.valid_for_uri?(URI("https://#{random_string}"))
+
+      expect(v).to eq(false)
+    end
+  end
+
+  describe 'acceptable_from_uri?' do
+    it 'will return false when passed nil' do
+      n = nil
+
+      a = cookie.acceptable_from_uri?(n)
+
+      expect(a).to eq(false)
+    end
+
+    it 'will return false if url without http(s) protocol is passed' do
+      generic_uri = random_string
+
+      v = cookie.acceptable_from_uri?(generic_uri)
+
+      expect(v).to eq(false)
+    end
+
+    it 'will return false if url with http(s) protocol is passed but has no host' do
+      protcol = 'http://'
+
+      v = cookie.acceptable_from_uri?(protcol)
+
+      expect(URI(protcol).is_a?(::URI::HTTP)).to eq(true)
+      expect(v).to eq(false)
+    end
+
+    it 'will return true if url with http(s) protocol is passed with a domain that matches the url domain' do
+      host = random_string
+      uri = "http://#{host}/#{random_string}"
+      cookie.domain = host
+
+      v = cookie.acceptable_from_uri?(uri)
+
+      expect(v).to eq(true)
+    end
+
+    it "will return domain.nil? if url with http(s) protocol is passed with a domain that doesn't match the url domain" do
+      uri = "http://#{random_string}/#{random_string}"
+      cookie.domain = rand(0..1) == 1 ? nil : random_string
+
+      v = cookie.acceptable_from_uri?(uri)
+
+      expect(v).to eq(cookie.domain.nil?)
+    end
```

Comment:
I think this test description need updated, or we're missing the https case. Let's leave this to a follow up PR if that's the case :+1: